state/application_test.go

```diff
@@ -1696,10 +1715,56 @@ func (s *ApplicationSuite) TestForceDoesNotAllowRemovingAnApplicationWithOffers(
 	op := s.mysql.DestroyOperation()
 	op.Force = true
 	err := s.State.ApplyOperation(op)
-	c.Assert(err, gc.ErrorMatches, `cannot destroy application "mysql": application is used by 1 offer`)
+	c.Assert(err, jc.ErrorIsNil)
 	assertOffersRef(c, s.State, "mysql", 1)
 }
 
+func (s *ApplicationSuite) TestForceRemoveApplicationsWithOffersThatHaveConnections(c *gc.C) {
```

Comment:
the title of this test isn't right
it implies that the app has offers to start out with, and we are testing the --force behaviour
but the test adds offers in a txn hook
so we typically call such tests TestFooRace which compliment TestFoo
so here you'd want to have
TestForceRemoveApplicationsWithOffersThatHaveConnections
and
TestForceRemoveApplicationsWithOffersThatHaveConnectionsRace

Reply:
An alternative would be to move this test and its non-Race variant into `applicationoffers_test.go`, where the fixutures are properly in place.

Reply:
Have pushed these tests into the other file whether the fixtures are already in place